### DISCUSSION TITLE
The difference between Outcome() and Done()

### DISCUSSION THREAD

**Discussion by amirmakramkolta** [elsa-workflows/elsa-core#2613] [Q&A] [answered]
Hi,
I am tried to make a custom activity and in the return statement I am tried to use `Outcome()` function
`return Outcome("your activity has been executed")` but when I use that my workflow ends with finish status on this activity

![1](https://user-images.githubusercontent.com/35640764/146776145-522aa4e4-7ed8-43f2-9e8d-ebce1fb9182e.PNG)

![2](https://user-images.githubusercontent.com/35640764/146776544-9a792320-6776-49dd-95ce-a72d665a235f.PNG)

If I used `Done()` instead of `Outcome()` my workflow works fine

**Comment by sfmskywalker** [comment, marked as answer]
Hi @amirmakramkolta 

`Done` is the same as `Outcome("Done")`.

However, `Done` also has an overload that allows you to provide output that is then used as input for the next activity:

`Done("Some output!")` which is the same as `Outcome("Done", "Some output!")`

Here are the implementations of both versions of `Done`:

```csharp
protected virtual OutcomeResult Done() => Outcome(OutcomeNames.Done);
protected virtual OutcomeResult Done(object? input) => Outcome(OutcomeNames.Done, input);
```

As you can see, `Done` is just a convenience method.

**Reply by amirmakramkolta** [reply]
So, How to get the input in the designer while writing with JavaScript Syntax

**Reply by sfmskywalker** [reply]
Using the `input` variable.
For example, if an activity returns this: `Done("Some output!")` then the next activity can access that output like this: 

```javascript
`The output is: ${input}`
```

**Reply by sfmskywalker** [reply]
More information on input & output can be found [here](https://elsa-workflows.github.io/elsa-core/docs/next/expressions/expressions-javascript#activity-output-elsa-21).

**Reply by amirmakramkolta** [reply]
Thank you that was helpful.
